lib/paginative/models/model_extension.rb

```diff
@@ -23,24 +26,88 @@ def self.with_id_from(id=0, limit=25)
 
       def self.with_field_from(field="", value="", limit=25, order="asc")
         order ||= "asc"
-        if field.is_a? Array
-          return raise "Wrong number of values. Expected 2, got #{value.try(:length)}. You must pass a value for each field that you are sorting by" unless value.is_a?(Array) && value.length == 2
-            # You can now pass in an array of 'field' params so that you can have a secondary sort order.
-            # This is important if your primary sort field could have duplicate values
-            primary_sort_field = field[0]
-            primary_value = value[0]
-            secondary_sort_field = field[1]
-            secondary_value = value[1]
-            # This allows us to pass in 2 different sort columns and still paginate correctly.
-            return self.order(sanitized_ordering(self.table_name, field, order)).where("#{self.table_name}.#{primary_sort_field} <= ? AND (#{self.table_name}.#{primary_sort_field} != ? OR #{self.table_name}.#{secondary_sort_field} < ?)", primary_value, primary_value, secondary_value) if order.try(:downcase) == "desc"
-
-            self.order(sanitized_ordering(self.table_name, field, order)).where("#{self.table_name}.#{primary_sort_field} >= ? AND (#{self.table_name}.#{primary_sort_field} != ? OR #{self.table_name}.#{secondary_sort_field} > ?)", primary_value, primary_value, secondary_value)
+
+        fields = Array.wrap(field).flatten
+        values = Array.wrap(value).flatten
+        zipped = fields.zip(values)
+        fields, values = prune_fields(zipped).transpose
+
+        q = self.all
+        if fields.present? && fields.any?
+          return raise "Too many" unless fields.length <= 2
+          return raise "Something...." unless values.length == fields.length
+
```

Comment:
Need to change these back to something useful...think they need to have coverage as well, actually.